File: demo/src/main/java/com/midas/demo/mvp/contract/HotTabContract.java
package com.midas.demo.mvp.contract;

import com.midas.demo.base.IPresenter;
import com.midas.demo.base.IView;
import com.midas.demo.mvp.model.bean.TabInfoBean;

public interface HotTabContract {

    interface View extends IView{
        /**
         * 设置 TabInfo
         */
        void setTabInfo(TabInfoBean tabInfoBean);

        void showError(String msg);
    }
    interface presenter extends IPresenter<View>{
        /**
         * 获取 TabInfo
         */
        void getTabInfo();
    }
}

File: app/src/main/java/com/midas/eye/mvp/contract/HotTabContract.kt
package com.midas.eye.mvp.contract

import com.midas.eye.base.IBaseView
import com.midas.eye.base.IPresenter
import com.midas.eye.mvp.model.bean.TabInfoBean

/**
 * Created by midas on 2019/05/28.
 * desc: 契约类
 */
interface HotTabContract {

    interface View: IBaseView {
        /**
         * 设置 TabInfo
         */
        fun setTabInfo(tabInfoBean: TabInfoBean)

        fun showError(errorMsg:String,errorCode:Int)
    }


    interface Presenter: IPresenter<View> {
        /**
         * 获取 TabInfo
         */
        fun getTabInfo()
    }
}